### ISSUE TITLE
Integrating sklearn with PySyft

### ISSUE THREAD

**Issue by MitanshuShaBa** [OpenMined/PySyft#5469]
## Feature Description
I am planning to add the following modules, classes and methods in `syft/lib/sklearn`:

**sklearn.linear_models**

Linear classifiers

- [ ] LogisticRegressionCV
- [ ] PassiveAggressiveClassifiers
- [ ] Perceptron
- [ ] RidgeClassifier
- [ ] RidgeClassifierCV
- [ ] SGDClassifier

Classical linear regressors

- [ ] Ridge
- [ ] RidgeCV
- [ ] SGDRegressor

Regressors with variable selection

- [ ] ElasticNet
- [ ] ElasticNetCV
- [ ] Lars
- [ ] LarsCV
- [ ] Lasso
- [ ] LassoCV
- [ ] LassoLars
- [ ] LassoLarsCV
- [ ] LassoLarsIC
- [ ] OrthogonalMatchingPursuit
- [ ] OrthogonalMatchingPursuitCV

Bayesian regressors

- [ ] ARDRegression
- [ ] BayesianRidge

Multi-task linear regressors with variable selection

- [ ] MultiTaskElasticNet
- [ ] MultiTaskElasticNetCV
- [ ] MultiTaskLasso
- [ ] MultiTaskLassoCV

Outlier-robust regressors

- [ ] HuberRegressor
- [ ] RANSACRegressor
- [ ] TheilSenRegressor

Generalized linear models (GLM) for regression

- [ ] PoissonRegressor
- [ ] TweedieRegressor
- [ ] GammaRegressor

Misc.

- [ ] PassiveAggresiveRegressor
- [ ] enet_path
- [ ] lars_path
- [ ] lars_path_gram
- [ ] lasso_path
- [ ] orthogonal_mp
- [ ] orthogonal_mp_gram
- [ ] ridge_regression

**sklearn.metrics**

Model Selection Interface

- [ ] check_scoring
- [ ] get_scorer
- [ ] make_scorer

Classification metrics

- [ ] average_precision_score
- [ ] balanced_accuracy_score
- [ ] brier_score_loss
- [ ] cohen_kappa_score
- [ ] dcg_score
- [ ] det_curve
- [ ] fbeta_score
- [ ] hamming_loss
- [ ] hinge_loss
- [ ] jaccard_score
- [ ] matthews_corrcoef
- [ ] ndcg_score
- [ ] precision_recall_fscore_support
- [ ] zero_one_loss

Regression  metrics

- [ ] explained_variance_score
- [ ] mean_squared_log_error
- [ ] median_absolute_error
- [ ] mean_absolute_percentage_error
- [ ] r2_score
- [ ] mean_poisson_deviance
- [ ] mean_gamma_deviance
- [ ] mean_tweedie_deviance

Multilabel ranking metrics

- [ ] coverage_error
- [ ] label_ranking_average_precision_score
- [ ] label_ranking_loss

Clustering metrics

- [ ] adjusted_mutual_info_score
- [ ] adjusted_rand_score
- [ ] calinski_harabasz_score
- [ ] davies_bouldin_score
- [ ] completeness_score
- [ ] cluster.contingency_matrix
- [ ] cluster.pair_confusion_matrix
- [ ] fowlkes_mallows_score
- [ ] homogeneity_completeness_v_measure
- [ ] homogeneity_score
- [ ] mutual_info_score
- [ ] normalized_mutual_info_score
- [ ] rand_score
- [ ] silhouette_score
- [ ] silhouette_samples
- [ ] v_measure_score

Biclustering metrics

- [ ] consensus_score

Pairwise metrics

- [ ] pairwise.additive_chi2_kernel
- [ ] pairwise.chi2_kernel
- [ ] pairwise.cosine_similarity
- [ ] pairwise.cosine_distances
- [ ] pairwise.distance_metrics
- [ ] pairwise.euclidean_distances
- [ ] pairwise.haversine_distances
- [ ] pairwise.kernel_metrics
- [ ] pairwise.laplacian_kernel
- [ ] pairwise.linear_kernel
- [ ] pairwise.manhattan_distances
- [ ] pairwise.nan_euclidean_distances
- [ ] pairwise.pairwise_kernels
- [ ] pairwise.polynomial_kernel
- [ ] pairwise.rbf_kernel
- [ ] pairwise.sigmoid_kernel
- [ ] pairwise.paired_euclidean_distances
- [ ] pairwise.paired_manhattan_distances
- [ ] pairwise.paired_cosine_distances
- [ ] pairwise.paired_distances
- [ ] pairwise_distances
- [ ] pairwise_distances_argmin
- [ ] pairwise_distances_argmin_min
- [ ] pairwise_distances_chunked

Plotting

- [ ] plot_confusion_matrix
- [ ] plot_det_curve
- [ ] plot_precision_recall_curve
- [ ] plot_roc_curve
- [ ] ConfusionMatrixDisplay
- [ ] DetCurveDisplay
- [ ] PrecisionRecallDisplay
- [ ] RocCurveDisplay

**Comment by Bhuvan-21** [comment]
Is this issue available to work on?

**Comment by MitanshuShaBa** [comment]
It was created for a GSoC Project.
